lighthouse-core/gather/gatherers/styles.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Gathers the active style and stylesheets used on a page.
+ * "Active" means that if the stylesheet is removed at a later time
+ * (before endStylesCollect is called), this gatherer will not include it.
+ */
+
+'use strict';
+
+const WebInspector = require('../../lib/web-inspector');
+const Gatherer = require('./gatherer');
+
+/**
+ * @param {!Artifacts} parseTree
+ * @return {!Array}
+ */
+function getCSSPropsInStyleSheet(parseTree) {
+  let results = [];
+
+  parseTree.traverseByType('declaration', function(node, index, parent) {
+    const [name, val] = node.toString().split(':').map(item => item.trim());
+    results.push({
+      property: {name, val},
+      declarationRange: node.declarationRange,
+      selector: parent.selectors.toString()
+    });
+  });
+
+  return results;
+}
+
+class Styles extends Gatherer {
+
+  constructor() {
+    super();
+    this._activeStyles = [];
+    this._onStyleSheetAdded = this.onStyleSheetAdded.bind(this);
+    this._onStyleSheetRemoved = this.onStyleSheetRemoved.bind(this);
+  }
```

Comment:
two overall things with this gatherer:
- need to do less work in `onStyleSheetAdded`. This is run during the page load as stylesheets come in. DBW doesn't look at traces yet, but it might at some point (and others may mix and match) so you don't want to be doing work as the page loads. Maybe the fetching and parsing of the stylesheet text be done in `endStylesCollect`?

- need to keep promise chains connected. `driver.sendCommand` returns a promise, so it should be brought in wherever possible to the existing gatherer chain (returned from the gatherer methods) so that execution ends up serialized between gatherers and so any errors are caught